src/lib/viewers/image/ImageViewer.js

```diff
@@ -165,6 +187,38 @@ class ImageViewer extends ImageBaseViewer {
         };
     }
 
+    /**
+     * Gets the viewport dimensions.
+     *
+     * @return {Object} the width & height of the viewport
+     */
+
+    getViewportDimensions() {
+        return {
+            width: this.wrapperEl.clientWidth - 2 * IMAGE_PADDING,
+            height: this.wrapperEl.clientHeight - 2 * IMAGE_PADDING,
+        };
+    }
+
+    /**
+     * Sets mode to be AnnotationMode.NONE if the zoomed image overflows the viewport.
+     *
+     * @return {void}
+     */
+
+    handleZoomEvent({ newScale }) {
+        const [width, height] = newScale;
+
+        const viewport = this.getViewportDimensions();
+
+        // We only set AnnotationMode to be NONE if the image overflows the viewport and the state is not explicit region creation
+        const { currentState } = this.annotationControlsFSM;
+        if (currentState !== AnnotationState.REGION && (width > viewport.width || height > viewport.height)) {
+            this.processAnnotationModeChange(this.annotationControlsFSM.transition(AnnotationInput.CANCEL));
+            this.annotator.toggleAnnotationMode(AnnotationMode.NONE);
```

Comment:
Seems like in `BaseViewer` we typically follow a `toggleAnnotationMode` followed by a `processAnnotationModeChange` -- should we just order it the same to be consistent?